profiling-ffi/src/profiles.rs

```diff
@@ -388,8 +389,9 @@ impl From<ProfileResult> for Result<(), String> {
 pub unsafe extern "C" fn ddog_prof_Profile_add(
     profile: *mut Profile,
     sample: Sample,
+    timestamp: Option<NonZeroI64>,
```

Comment:
Just to doublecheck -- this gets translated to a `int64_t` in the C headers; does that mean `None` gets represented by `0`?

Reply:
It's specifically documented in the std https://doc.rust-lang.org/stable/std/num/struct.NonZeroI64.html

Reply:
> NonZeroI64 is guaranteed to have the same layout and bit validity as i64 with the exception that 0 is not a valid instance. Option<NonZeroI64> is guaranteed to be compatible with i64, including in FFI.

Is that what this means? Because... it's rather cryptic lol.

Reply:
yes I think it's quite explicit